stravalib/model.py

```diff
@@ -412,37 +654,50 @@ def is_authenticated_athlete(self):
             else:
                 # We need to check this athlete's id matches the authenticated athlete's id
                 authenticated_athlete = self.authenticated_athlete
+                assert authenticated_athlete is not None
```

Comment:
Here are we sure the authenticated_athlete will always be not None? Maybe it's clear that it will always be not None or perhaps it should return False if it is None?

Reply:
ok this also is making my brain 🤯 a bit 😆 authenticated_athlete is a bool

i implemented this fix:

```python
if authenticated_athlete is None:
     return False
```
 
which i think makes more sense. i'm struggling to know whether there are edge cases where we could encounter  unexpected None values and where they would be. but i do think here it makes sense to simply return a False given it's an authentication check so if the code has gotten that far... we'd assume `false` is a safe response..